service-impl/cache-service-impl/simple-cache-service/src/main/java/uk/gov/gchq/palisade/cache/service/impl/SimpleCacheObject.java

```diff
@@ -40,17 +39,34 @@
      */
     private final Optional<byte[]> value;
 
+    /**
+     * Extra information about this entry and it's retrieval.
+     */
+    private final Optional<CacheMetadata> metadata;
+
     /**
      * Create a cache object.
      *
      * @param valueClass the type of the value being cached
      * @param value      the optional cache value, may be empty if no valid entry is present
+     * @param metadata   information about the cache entry
      */
-    public SimpleCacheObject(final Class<?> valueClass, final Optional<byte[]> value) {
+    public SimpleCacheObject(final Class<?> valueClass, final Optional<byte[]> value, final Optional<CacheMetadata> metadata) {
```

Comment:
I don't think the metadata is needed in the constructor if you have a set metadata method, given majority of the time this is set to Optional.empty()

Reply:
Done